hook!-full-game/src/Player/Hook/Skills/Base.gd

```diff
@@ -25,16 +24,33 @@ func _unhandled_input(event: InputEvent) -> void:
 
 
 func _hook() -> void:
+	if not hook.cooldown.is_stopped():
+		return
+	_snap_hook_arrow_tip()
+	_disable_hook_aim()
+	_handle_hook_target()
+	_trigger_hook_cooldown()
```

Comment:
You can replace with `hook.cooldown.start()` straight, the method only adds complexity

Reply:
This is the `Base` skill, right? I think we might want to have everything that happens in the `_hook` encapsulated so we can extend it reliably later. I tried to do that by creating an `Attack` skill, and basically, I needed to duplicate the whole code on `_hook` when overriding it to change just the line that triggers the hook cooldown.

Same goes for the `_disable_hook_aim` we may want to change how each skill disable the hook aim, e.g. if there are multiple enemies.

Reply:
There's no need to worry about these details here, as mentioned in the 1st comment Nathan left, the skills system will change to the same HFSM as in `Player`, we can merge as is as I'll need to go over this later anyway.